src/Sylius/Bundle/ProductBundle/Doctrine/ORM/ProductAttributeValueRepository.php

```diff
@@ -27,8 +29,18 @@ class ProductAttributeValueRepository extends EntityRepository implements Produc
 {
     public function findByJsonChoiceKey(string $choiceKey): array
     {
-        return $this->createQueryBuilder('o')
-            ->andWhere('o.json LIKE :key')
+        $connection = $this->getEntityManager()->getConnection();
+        $isPostgres = $this->isPostgreSQLPlatform($connection);
+
+        $queryBuilder = $this->createQueryBuilder('o');
+
+        if ($isPostgres) {
```

Comment:
```suggestion
        $queryBuilder = $this->createQueryBuilder('o');

        if ($this->isPostgresSQLPlatform()) {
```
Don't see a point in getting and passing the connection in here if it's completely dislocated with the rest of the method, just put it in the `::isPostgresSQLPlatform`